sim/src/kernels/computeEta.cl

```diff
@@ -4,64 +4,87 @@
 #ifndef __OPENCL_VERSION__
 #define __kernel
 #define __global
-#define local
+#define __local
 #define CLK_LOCAL_MEM_FENCE
 #endif
 
-/*
-Computes Eta at the center of the (non-ghost) cell. Eta is the sea surface deviation away from the equilibrium depth (H).
-Input:
-  - args.nx, args.ny: Internal grid size (nx, ny), i.e. not including grid points at the outside of ghost cells.
-  - U: Depth averaged velocity in the x-direction. Defined at western and eastern cell edges, excluding those of the southern
-       and northern ghost cells. Size: (nx + 2, ny - 1).
-  - V: Depth averaged velocity in the y-direction. Defined at southern and northern cell edges, excluding those of the western
-       and eastern ghost cells. Size: (nx - 1, ny + 2).
-Input+output:
-  - eta: Values defined at the center of each cell, including ghost cells. Size: (nx + 1, ny + 1).
-*/
-__kernel void computeEta (
-    __global const float *U,
-    __global const float *V,
-    __global float *eta,
-    computeEta_args args)
-{
-    const int nx = args.nx;
-    const int ny = args.ny;
+/**
+  * Kernel that evolves eta one step in time.
+  */
+__kernel void computeEta(
+		computeEta_args args,
+        //Data
+        __global float* U_ptr, int U_pitch,
+        __global float* V_ptr, int V_pitch,
+        __global float* eta_ptr, int eta_pitch) {
 
-    // work item within group
-	const unsigned int lx = get_local_id(0);
-	const unsigned int ly = get_local_id(1);
+    //Index of thread within block
+    const int tx = get_local_id(0);
```

Comment:
Again, this kind of change makes it very difficult to keep one coherent code. Bugfixes in the python version will not propagate easily here, and vise versa.